### PR TITLE
EWPP-921 - Drupal 9 compatibility fixes

### PR DESCRIPTION
**Pull request by GilNovacomm**
### Description

Drupal 9 compatibility update.

### Change log

- Added: Minimum Drupal 9.1 version, new environment variables for Drone, updated definition on README.
- Changed: Updated required modules
- Deprecated:
- Removed: PHP 7.2
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -69,8 +73,7 @@
         "patches": {
             "drupal/content_lock": {
                 "https://www.drupal.org/project/content_lock/issues/2919019": "https://www.drupal.org/files/issues/content_lock-caching-invalidation-2919019-3.patch",
-                "https://www.drupal.org/project/content_lock/issues/2956625": "https://www.drupal.org/files/issues/2018-03-28/error-on-locked-content-by-deleted-user-2956625.2.patch",
-                "https://www.drupal.org/project/content_lock/issues/2949198": "https://www.drupal.org/files/issues/content_lock-unlocking-already-locked-2949198-9.patch"
+                "https://www.drupal.org/project/content_lock/issues/2949198": "https://git.drupalcode.org/project/content_lock/-/merge_requests/3.patch"
```
Comment (drishu) [inline composer.json:76]: This patch needs more work, see https://www.drupal.org/pift-ci-job/2078065

**Review by drishu — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (saidatom) [reply composer.json:76]: Fixed - https://www.drupal.org/pift-ci-job/2092927

**Comment by saidatom** [comment]
Module "[ec-europa/oe-poetry-client](https://github.com/ec-europa/oe-poetry-client)" prevent upgrade to Drupal 9.

File: modules/oe_editorial_content_lock/oe_editorial_content_lock.info.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
-name: 'OpenEuropa Editorial Content Lock'
-description: 'Lock content to prevent concurrent editing.'
+name: OpenEuropa Editorial Content Lock
+description: Lock content to prevent concurrent editing.
```
Comment (22Alexandra) [inline modules/oe_editorial_content_lock/oe_editorial_content_lock.info.yml:2]: Single quotes are actually needed when there is more than one word.

File: modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/modules/oe_editorial_corporate_workflow_translation_poetry/oe_editorial_corporate_workflow_translation_poetry.info.yml [hunk 3]
```diff
@@ -1,8 +1,8 @@
-name: 'OpenEuropa Editorial Corporate Workflow Translation Poetry'
-description: 'Integrates the editorial workflow with OE Translation Poetry'
+name: OpenEuropa Editorial Corporate Workflow Translation Poetry
+description: Integrates the editorial workflow with OE Translation Poetry
```
Comment (22Alexandra) [inline modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/modules/oe_editorial_corporate_workflow_translation_poetry/oe_editorial_corporate_workflow_translation_poetry.info.yml:2]: Single quotes are actually needed when there is more than one word.

File: modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/oe_editorial_corporate_workflow_translation.info.yml [hunk 4]
```diff
@@ -1,8 +1,8 @@
-name: 'OpenEuropa Editorial Corporate Workflow Translation'
-description: 'Integrates the editorial workflow with OE Translation'
+name: OpenEuropa Editorial Corporate Workflow Translation
+description: Integrates the editorial workflow with OE Translation
```
Comment (22Alexandra) [inline modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/oe_editorial_corporate_workflow_translation.info.yml:2]: Single quotes are actually needed when there is more than one word.

File: modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/src/EventSubscriber/CorporateWorkflowEntitySourceSubscriber.php [hunk 5]
```diff
@@ -8,6 +8,12 @@
 use Drupal\Core\Entity\ContentEntityInterface;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\oe_translation\Event\ContentEntitySourceEntityEvent;
+/**
```
Comment (22Alexandra) [inline modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/src/EventSubscriber/CorporateWorkflowEntitySourceSubscriber.php:11]: Please add a space between use statement block and class annotation.

File: modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/tests/modules/oe_editorial_corporate_workflow_translation_test/oe_editorial_corporate_workflow_translation_test.info.yml [hunk 6]
```diff
@@ -1,8 +1,8 @@
-name: 'OpenEuropa Editorial Corporate Workflow Translation Test'
-description: 'Test module for the OpenEuropa Editorial Corporate Workflow Translation'
+name: OpenEuropa Editorial Corporate Workflow Translation Test
+description: Test module for the OpenEuropa Editorial Corporate Workflow Translation
```
Comment (22Alexandra) [inline modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/tests/modules/oe_editorial_corporate_workflow_translation_test/oe_editorial_corporate_workflow_translation_test.info.yml:2]: The same as for the others. Please add the single quotes back where it's been removed and add them where missing.

File: modules/oe_editorial_unpublish/src/Event/UnpublishStatesEvent.php [hunk 7]
```diff
@@ -5,6 +5,12 @@
 namespace Drupal\oe_editorial_unpublish\Event;
 
 use Drupal\Core\Entity\ContentEntityInterface;
+/**
```
Comment (22Alexandra) [inline modules/oe_editorial_unpublish/src/Event/UnpublishStatesEvent.php:8]: Add a space between use statement and class annotation.

**Review by 22Alexandra — CHANGES_REQUESTED** [review]
All the tests placed directly in a `tests` folder should be moved under a `src` folder (e.g. `modules/oe_editorial_unpublish/tests` -> `modules/oe_editorial_unpublish/tests/src`).

PR needs to be rebased.

File: composer.json [hunk 8]
```diff
@@ -6,34 +6,29 @@
...
+        "drupal/config_devel": "^1.2",
+        "drupal/content_lock": "^2.2",
+        "drupal/drupal-extension": "^4.0",
+        "drupal/entity_version": "^1.0-beta8",
         "drupal/address": "^1.8",
-        "openeuropa/oe_paragraphs": "~1.6.0",
-        "drush/drush": "~9.0@stable",
-        "ec-europa/oe-poetry-client": "^0.4",
-        "guzzlehttp/guzzle": "^6.3",
+        "openeuropa/oe_paragraphs": "dev-master",
```
Comment (brummbar) [inline composer.json:20]: Paragraphs has just been released, so we can update.

**Review by brummbar — CHANGES_REQUESTED** [review]
By running the update check , there's an inconsistent return type declaration in /modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/modules/oe_editorial_corporate_workflow_translation_poetry/tests/src/Functional/PoetryTranslationsTest.php

**Review by brummbar — APPROVED** [review]

**Review by nagyad — APPROVED** [review]

**Review by upchuk — APPROVED** [review]

**Review by brummbar — APPROVED** [review]

**Pull request merged**